code/game/objects/items/RPD.dm

```diff
@@ -252,9 +254,56 @@ GLOBAL_LIST_INIT(fluid_duct_recipes, list(
 	spark_system = null
 	return ..()
 
+/obj/item/pipe_dispenser/examine(mob/user)
+	. = ..()
+	. += "You can scroll your mouse wheel to change the piping layer."
+
+/obj/item/pipe_dispenser/equipped(mob/user, slot, initial)
+	. = ..()
+	if(slot == ITEM_SLOT_HANDS)
+		RegisterSignal(user, COMSIG_MOUSE_SCROLL_ON, .proc/mouse_wheeled)
+	else
+		UnregisterSignal(user, COMSIG_MOUSE_SCROLL_ON)
+
+/obj/item/pipe_dispenser/dropped(mob/user, silent)
+	UnregisterSignal(user, COMSIG_MOUSE_SCROLL_ON)
+	return ..()
+
+/obj/item/pipe_dispenser/cyborg_unequip(mob/user)
+	UnregisterSignal(user, COMSIG_MOUSE_SCROLL_ON)
+	return ..()
+
+/obj/item/pipe_dispenser/pre_attack(atom/target, mob/user, params)
+	if(istype(target, /obj/item/rpd_upgrade/unwrench))
```

Comment:
why is this one specifically on unwrench?

Reply:
Direct port from /tg/, looks like the RPD main proc also uses pre_attack, this part is so that cyborgs can install the upgrade with their RPD.